src/Bicep.Cli/CommandLine/Arguments/BuildArguments.cs

```diff
@@ -1,31 +1,68 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
-using System;
-using System.Collections.Generic;
-using System.Collections.Immutable;
-using System.Linq;
 
 namespace Bicep.Cli.CommandLine.Arguments
 {
     public class BuildArguments : ArgumentsBase
     {
-        private static bool IsStdOutArgument(string argument)
-            => StringComparer.OrdinalIgnoreCase.Equals(argument, CliConstants.ArgumentStdOut);
-
-        public BuildArguments(IEnumerable<string> arguments)
+        public BuildArguments(string[] args)
         {
-            this.OutputToStdOut = arguments.Where(IsStdOutArgument).Any();
-            this.Files = arguments.Where(f => !IsStdOutArgument(f)).ToImmutableArray();
+            for (var i = 0; i < args.Length; i++)
+            {
+                switch (args[i].ToLowerInvariant()) {
+                    case "--stdout":
+                        this.OutputToStdOut = true;
+                        break;
+                    case "--outdir":
+                        if (args.Length == i + 1)
+                        {
+                            throw new CommandLineException($"The --outdir parameter expects an argument");
+                        }
+                        if (this.OutputDir is not null)
+                        {
+                            throw new CommandLineException($"The --outdir parameter cannot be specified twice");
+                        }
+                        this.OutputDir = args[i + 1];
+                        i++;
+                        break;
+                    case "--outfile":
```

Comment:
>outfile [](start = 28, length = 7)

Do we have a check somewhere that blocks usage of --outdir together with --outfile?

Reply:
No - I'll add this